FusionIIIT/applications/hostel_management/models.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+from django.db import models
+from django.contrib.auth.models import User
+from applications.globals.models import ExtraInfo, Staff, Faculty
+from applications.academic_information.models import Student
+from applications.complaint_system.models import Caretaker
+from django.utils import timezone
+
+
+class HostelManagementConstants:
+    ROOM_STATUS = (
+        ('Booked', 'Booked'),
+        ('CheckedIn', 'Checked In'),
+        ('Available', 'Available'),
+        ('UnderMaintenance', 'Under Maintenance'),
+        )
+
+    DAYS_OF_WEEK = (
+            (0, 'Monday'),
+            (1, 'Tuesday'),
+            (2, 'Wednesday'),
+            (3, 'Thursday'),
+            (4, 'Friday'),
+            (5, 'Saturday'),
+            (6, 'Sunday')
+        )
+
+    BOOKING_STATUS = (
+    ("Confirmed" , 'Confirmed'),
+    ("Pending" , 'Pending'),
+    ("Rejected" , 'Rejected'),
+    ("Canceled" , 'Canceled'),
+    ("CancelRequested" , 'Cancel Requested'),
+    ("CheckedIn" , 'Checked In'),
+    ("Complete", 'Complete'),
+    ("Forward", 'Forward')
+    )
+
+
+
+class Hall(models.Model):
+    hall_id = models.CharField(max_length=10)
+    hall_name = models.CharField(max_length=50)
+    max_accomodation = models.IntegerField(default=0)
+    number_students = models.PositiveIntegerField(default=0)
+    hall_caretaker = models.ManyToManyField(Caretaker, on_delete=models.CASCADE)
+    hall_warden = models.ManyToManyField(Faculty, on_delete=models.CASCADE)
```

Comment:
Similar to above, instead of this field, create a new model  with name `HallWarden` with the following fields:

* faculty --> F.K. to Faculty model.
* hall --> F.K. to Hall model.